patches/server/0731-Add-System.out.println-catcher.patch

```diff
@@ -0,0 +1,106 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: underscore11code <minecrafter11mrt@gmail.com>
+Date: Fri, 23 Jul 2021 23:01:42 -0700
+Subject: [PATCH] Add System.out.println catcher
+
+
+diff --git a/src/main/java/io/papermc/paper/logging/SysoutCatcher.java b/src/main/java/io/papermc/paper/logging/SysoutCatcher.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..b55228f1e5f3fed43f19e83dbdf77e93258c4cd8
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/logging/SysoutCatcher.java
+@@ -0,0 +1,74 @@
++package io.papermc.paper.logging;
++
++import com.google.common.collect.ImmutableSet;
++import org.apache.commons.lang.StringUtils;
++import org.bukkit.Bukkit;
++import org.bukkit.plugin.Plugin;
++import org.bukkit.plugin.java.JavaPlugin;
++
++import java.io.PrintStream;
++import java.util.Set;
++
++public final class SysoutCatcher {
++    private static final String OUT_PREFIX = "[STDOUT] ";
++    private static final String ERR_PREFIX = "[STDERR] ";
++    private final StackWalker stackWalker = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE);
++    private final Set<String> blacklistedPlugins;
++
++    public SysoutCatcher() {
++        final String rawBlacklist = System.getProperty("io.papermc.paper.suppress.sout.nags");
++        if (StringUtils.isNotBlank(rawBlacklist)) {
++            blacklistedPlugins = ImmutableSet.<String>builder().add(rawBlacklist.split(",")).build();
++        } else blacklistedPlugins = ImmutableSet.of();
++
++        // Listen to System.out.println
++        PrintStream wrappedOut = new PrintStream(System.out) {
```

Comment:
I'd prefer if these anonymous classes were replaced with a private class and use variables for the different constants used.

Reply:
All right, I've done what I assume you meant, lmk how this looks.